hoot-core/src/main/cpp/hoot/core/io/schema/LongIntegerFieldDefinition.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2015 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#ifndef LONGINTEGERFIELDDEFINITION_H
+#define LONGINTEGERFIELDDEFINITION_H
+
+// Standard
+#include <set>
+
+#include "FieldDefinition.h"
+
+namespace hoot
+{
+using namespace std;
+
+class LongIntegerFieldDefinition : public FieldDefinition
+{
+public:
+  LongIntegerFieldDefinition();
+
+  void addEnumeratedValue(int v) { _enumeratedValues.insert(v); }
+
+  virtual QVariant getDefaultValue() const;
+
+  virtual QVariant::Type getType() const { return QVariant::LongLong; }
+
+  virtual bool hasDefaultValue() const;
+
+  bool hasEnumeratedValue(int v) { return _enumeratedValues.find(v) != _enumeratedValues.end(); }
+
+  void setDefaultValue(double v) { _defaultValue = v; }
```

Comment:
AND... I know this is a copy from IntegerFieldDefinition.h - but I think the params for these three functions (setDefaultValue, setMaxValue, setMinValue) should all be type ```long long```, not double.

Unless I'm misunderstanding something.